houndigrade/test_cli.py

```diff
@@ -0,0 +1,224 @@
+
+        mock_glob_glob.side_effect = mock_glob_side_effect
+
+        mock_with_conn = mock_connection.__enter__.return_value
+        mock_producer = mock_with_conn.Producer.return_value
+        mock_pub = mock_producer.publish
+        mock_pub.return_value = True
+
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            pathlib.Path('{}/xvdf1'.format(drive_path)).mkdir(parents=True,
+                                                              exist_ok=True)
+            result = runner.invoke(main,
+                                   ['-c', cloud, '--debug', '-t', image_id,
+                                    drive_path])
+
+        self.assertTrue(mock_subprocess_run.called)
+        self.assertEqual(result.exit_code, 0)
+        self.assertIn(
+            '"status": "No release files found on ./dev/xvdf1"', result.output)
+        self.assertIn(
+            '"status": "No release files found on ./dev/xvdf2"', result.output)
+
+    @patch('cli.Connection')
+    @patch('cli.glob.glob')
+    @patch('cli.subprocess.run')
+    def test_failed_mount(self, mock_subprocess_run, mock_glob_glob, mock_con):
+        image_id = 'ami-123456789'
+        drive_path = './dev/xvdf'
+
+        e = CalledProcessError(1, 'mount', stderr='Mount failed.')
+
+        mock_subprocess_run.side_effect = e
+
+        def mock_glob_side_effect(pattern):
+            return ['./dev/xvdf1']
+
+        mock_glob_glob.side_effect = mock_glob_side_effect
+
+        mock_with_conn = mock_con.return_value__enter__.return_value
+        mock_producer = mock_with_conn.Producer.return_value
+        mock_pub = mock_producer.publish
+        mock_pub.return_value = True
+
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            pathlib.Path('{}/xvdf1'.format(drive_path)).mkdir(parents=True,
+                                                              exist_ok=True)
+            result = runner.invoke(main, ['-t', image_id, drive_path])
+
+        self.assertTrue(mock_subprocess_run.called)
+        self.assertEqual(result.exit_code, 0)
+        self.assertEqual('Mount failed.',
+                      mock_con.mock_calls[3][1][0]['facts']['./dev/xvdf'][
+                          './dev/xvdf1'][0]['error'])
+
+    @staticmethod
+    def prep_fs(drive_path):
+        pathlib.Path('{}/xvdf1/etc'.format(drive_path)).mkdir(parents=True,
+                                                              exist_ok=True)
+        pathlib.Path('{}/xvdf2/etc'.format(drive_path)).mkdir(parents=True,
+                                                              exist_ok=True)
+
+        with open('{}/xvdf1/etc/redhat-release'.format(drive_path), 'w') as f:
+            f.write('Red Hat Enterprise Linux Server release 7.4 (Maipo)\n')
+        with open('{}/xvdf1/etc/os-release'.format(drive_path), 'w') as f:
+            f.write('NAME=\"Red Hat Enterprise Linux Server\"\nVERSION=\"7.4 '
```

Comment:
Are all those double-quote escapes really necessary since the string is defined with single-quotes?

Reply:
Also, what would you think about making this a triple-quote string so you don't have to escape the newlines?

Maybe shove the whole thing into a variable using triple-quotes and pass it through [textwrap.dedent](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) before `f.write`ing it. I think that might make this more legible.